src/sentry/search/snuba/executors.py

```diff
@@ -334,6 +334,10 @@ def snuba_search(
                 if gc != GroupCategory.PROFILE
                 or features.has("organizations:issue-platform", organization)
             }
+
+        if not features.has("organizations:performance-issues-search", organization):
+            group_categories.discard(GroupCategory.PERFORMANCE)
```

Comment:
I think just adding this would've done the job instead of conditionally adding extra filters.

Reply:
Yeah, might be simpler to just leave this in place and avoid the other step? We might actually leave this flag in place for a while, depending on how ST wants to handle things.

Reply:
Yeah, gonna clean it up to make it easier to remove it later.